src/tickets/tickets.service.ts

```diff
@@ -0,0 +1,95 @@
+import { httpErrors } from '@fastify/sensible'
+import { sql } from 'drizzle-orm'
+import { FastifyInstance, FastifyRequest } from 'fastify'
+import { TCreateTicketSchema } from './schema/create-ticket.schema'
+import { getUserBySessionId } from '../auth/auth.service'
+import { TUpdateTicketSchema } from './schema/update-ticket.schema'
+
+export const getTicketsHandler = async (app: FastifyInstance) => {
+  const response = app.db.query.tickets.findMany({
+    with: { attachments: true },
+  })
+
+  return response
+}
+
+export const getTicketHandler = async (app: FastifyInstance, uuid: string) => {
+  const response = await app.db.execute(sql`
+    SELECT * FROM tickets
+    WHERE uuid = ${uuid}
+    `)
+
+  if (!response.rows.length) {
+    throw httpErrors.notFound('Ticket not found')
+  }
+
+  return response.rows
+}
+
+export const postTicketHandler = async (
+  app: FastifyInstance,
+  request: FastifyRequest
+) => {
+  const sessionId = request.session?.sessionId
+  const user = await getUserBySessionId(app.db, sessionId)
```

Comment:
that should be mostly done in some middleware, pre request hook, controller-
so you have something like (arbitrary example

```
app.post(/posts/id, { auth: true, role: 'Admin' }, (req)=> {
...bla bla and you u already expect the user to be present and be admin here

}) 

```
and you don't handle the